### PR TITLE
Ensure native notebooks can be used without Python or Python extension

### PR DESCRIPTION
**Pull request by DonJayamanne**
For #1074
Separate test where Python extension is not installed.

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -560,7 +564,27 @@ jobs:
           github_token: ${{ secrets.GITHUB_TOKEN }}
           report_paths: ${{ env.TEST_RESULTS_GLOB }}
           check_name: Notebook Test Report
-        if: (steps.test_notebook_vscode.outcome == 'failure' || steps.test_notebook_vscode.outcome == 'failure') && failure()
+        if: steps.test_notebook_vscode.outcome == 'failure' && failure()
+
+      - name: Run Native Notebook with VSCode & Jupyter (without Python)
```
Comment (DonJayamanne) [inline .github/workflows/pr.yml:569]: Specific tests without python extension & python.
This way we never break & gets tested on CI.

File: src/client/datascience/notebook/kernelProvider.ts [hunk 2]
```diff
@@ -135,7 +137,7 @@ export class VSCodeKernelPickerProvider implements NotebookKernelProvider {
                 undefined,
                 token
             ),
-            this.interpreterService.getActiveInterpreter(document.uri)
+            isPythonNb ? this.interpreterService.getActiveInterpreter(document.uri) : Promise.resolve(undefined)
```
Comment (rchiodo) [inline src/client/datascience/notebook/kernelProvider.ts:140]: Won't this still ask if the notebook is a python notebook? What we really want is to do nothing with the interpreters unless the python extension is installed. See IPythonExtensionChecker

**Review by rchiodo — CHANGES_REQUESTED** [review]
⏲️

File: src/client/datascience/notebook/kernelProvider.ts (hunk 2 above)
Comment (DonJayamanne) [reply src/client/datascience/notebook/kernelProvider.ts:140]: Oh ok, thanks didn't expect that.
Makes sense

**Review by DavidKutu — APPROVED** [review]

File: src/client/datascience/notebook/kernelProvider.ts (hunk 2 above)
Comment (rchiodo) [reply src/client/datascience/notebook/kernelProvider.ts:140]: This doesn't seem to be resolved yet.

Comment (DonJayamanne) [reply src/client/datascience/notebook/kernelProvider.ts:140]: Damn it, yes

Comment (DonJayamanne) [reply src/client/datascience/notebook/kernelProvider.ts]: Done

**Comment by DonJayamanne** [comment]
Need to fix the failing test.

**Review by rchiodo — APPROVED** [review]

**Pull request merged**